File: apps/macos/KawaCAD/Sources/KawaCADApp/Features/Document/Selectors/DocumentWindowPresentation.swift
import Foundation

struct DocumentWindowPresentation: Equatable {
  let documentURL: URL?
  let isDocumentEdited: Bool

  var title: String {
    guard let documentURL else {
      return AppStrings.tr("window.title.unsaved", AppStrings.tr("app.document.untitled"))
    }
    return documentURL.lastPathComponent
  }

  var accessibilityLabel: String {
    let editedState = AppStrings.tr(
      isDocumentEdited ? "window.accessibility.edited" : "window.accessibility.saved"
    )
    if let documentURL {
      return AppStrings.tr(
        "window.accessibility.saved_document",
        documentURL.lastPathComponent,
        editedState
      )
    }
    return AppStrings.tr(
      "window.accessibility.unsaved_document", AppStrings.tr("app.document.untitled"),
      editedState)
  }
}

File: apps/tauri/KawaCAD/src/features/workspace/selectors/documentWindowPresentation.ts
import { appStrings } from "@/localization";

export function documentDisplayName(path: string | undefined) {
  return path?.split(/[\\/]/).pop() || appStrings.app.untitled;
}

export function documentWindowPresentation(path: string | undefined, isDirty: boolean) {
  const displayName = documentDisplayName(path);
  if (!path)
    return {
      title: `${displayName} — ${appStrings.app.unsaved}`,
      accessibilityLabel: `${displayName}、${isDirty ? appStrings.app.unsavedChanges : appStrings.app.unsaved}`,
    };
  return {
    title: displayName,
    accessibilityLabel: `${displayName}、${isDirty ? appStrings.app.unsavedChanges : appStrings.app.savedState}`,
  };
}
